src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -44,6 +44,29 @@ public function getRelationInstance($field)
         return $relation_model->{$related_method}();
     }
 
+    /**
+     * Get the fields with specific relation types.
+     *
+     * @param string|array $relation_types
+     *
+     * @return array The fields with corresponding relation types.
+     */
+    public function getFieldsWithRelationType($relation_types): array
```

Comment:
Uuu... return types. Sexy! 😄

Reply:
Will this conflict with any of our supported php versions ?